packages/tailwindcss/src/utilities.ts

```diff
@@ -683,20 +700,26 @@ export function createUtilities(theme: Theme) {
     },
     themeKeys: ['--grid-column'],
     handle: (value) => [decl('grid-column', value)],
+    staticValues: {
+      auto: [decl('grid-column', 'auto')],
+    },
   })
-  staticUtility('col-span-full', [['grid-column', '1 / -1']])
+  // Ensure discoverability for `col-auto` after converting to fallback
+  suggest('col', () => [{ values: ['auto'], valueThemeKeys: ['--grid-column'] }])
```

Comment:
I wonder if we can make these suggestions more automatic as part of the `functionalUtility` helper because we know the root, we know the values (keys of `staticValues`) and we know the `valueThemeKeys`.

Should simplify some of the additional code and ensure everything is in sync when we update some static values. Thoughts?

Reply:
Can't provide a suggestion to code that didn't change (thanks GitHub!), but something like this should work I think:
```ts
    if (desc.staticValues) {
      let values = Object.keys(desc.staticValues)
      suggest(classRoot, () => [{ values, valueThemeKeys: desc.themeKeys ?? [] }])
    }
```
at the end of the `functionalUtility` function

Reply:
Hmmmm, right now though, the `functionalUtility` does not provide any suggestions on its own and I think that's a good thing: If we ever decide to double-down on the browser build, we'd want a convenient way to compile the suggestions away and then this current setup will become useful. That said for the previous `staticUtility()` helper, suggestions _are provided by default_ so it wasn't very consistent to begin with so I'm fine with either approach.

Reply:
> right now though, the `functionalUtility` does not provide any suggestions on its own

What do you mean by this exactly? We do have this already:

https://github.com/tailwindlabs/tailwindcss/blob/340b59dcde2401f03f1edecf1c1332190adc9d44/packages/tailwindcss/src/utilities.ts#L449-L456

Which is part of the `functionalUtility` function when being called.

Reply:
Oh I see, interesting! Then I guess I am wrong and it should just work. no idea why namespaces keys aren't added automatically then too, though 🤔

Reply:
@RobinMalfait updated!